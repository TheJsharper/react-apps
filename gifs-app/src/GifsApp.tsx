
import GifList from './gifs/components/GifList';
import PreviousSearches from './gifs/components/previousSearches';
import { useGif } from './gifs/hooks/useGif';
import GifsHeader from './shared/components/GifsHeader';
import SearchBar from './shared/components/SearchBar';
const GifsApp = () => {
 const  { gifs, previousSearches, handleSearchClick, handleSearchSubmit } = useGif();
    return (
        <>
            {/* Header */}
            <GifsHeader title='Search Gifs' description='Find the best GIFs for your needs!' />
            {/* Search Form */}
            <SearchBar onSearchSubmit={handleSearchSubmit} placeholder="Search a gif here..." />
            {/* List of Gifs */}
            <PreviousSearches title="Previous Searches" searches={previousSearches}
                onSearchClick={handleSearchClick} />
            <GifList gifs={gifs} />
        </>)
}



export default GifsApp
