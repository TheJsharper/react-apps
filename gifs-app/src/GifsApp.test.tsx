import { describe, expect, it } from 'vitest';
import GifsApp from './GifsApp';
import { render } from '@testing-library/react';
describe('GifsApp', () => {
  it('should render correctly', () => {
    const { container } = render(<GifsApp />);
    expect(container).toMatchSnapshot();
  });
});