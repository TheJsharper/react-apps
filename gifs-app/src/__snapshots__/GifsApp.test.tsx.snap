// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > should render correctly 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Search Gifs
    </h1>
    <p>
      Find the best GIFs for your needs!
    </p>
  </div>
  <form
    class="search-container"
  >
    <input
      placeholder="Search a gif here..."
      type="text"
      value=""
    />
    <button>
       Search
    </button>
  </form>
  <div
    class="previous-searches"
  >
    <h2>
      Previous Searches
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
